tendenci/apps/directories/views.py

```diff
@@ -56,8 +56,41 @@ def details(request, slug=None, template_name="directories/view.html"):
 
 @is_enabled('directories')
 def search(request, template_name="directories/search.html"):
-    filters = get_query_filters(request.user, 'directories.view_directory')
+    user = request.user
+    username = request.user.username
+    filters = get_query_filters(user, 'directories.view_directory')
     directories = Directory.objects.filter(filters).distinct()
+
+    # Checking for view and change directory permission (staff)
+    can_view_directories = has_perm(user, 'directories.view_directory')
+    can_change_directories = has_perm(user, 'directories.change_directory')
+
+    can_view_pend = False
```

Comment:
This looks really nice. I'd like to be able to use this for other views as well. Can we expose the permission check as a Directory.is_editable_by(user) method within the directory model?

Reply:
Newest commit shows changes to:
`app/directories/utils.py` 

- can_view_pending_directories function to allow it to be callable in other models
- Imported the has_perm function to allow checking permissions against a user

and 

`app/directories/views.py`

- Condensed down code for a more pythonic flow

-